linkerd/app/gateway/src/lib.rs

```diff
@@ -158,11 +152,13 @@ where
                         .stack
                         .layer(metrics::StackLabels::inbound("tcp", "gateway")),
                 )
-                .push(svc::layer::mk(svc::SpawnReady::new))
-                .push(svc::FailFast::layer("TCP Gateway", dispatch_timeout))
-                .push_spawn_buffer(buffer_capacity),
+                .push_buffer(
+                    "TCP Gatweay",
+                    inbound_config.tcp_server_buffer.capacity,
+                    inbound_config.tcp_server_buffer.failfast_timeout,
```

Comment:
@olix0r wdyt of having this take a `BufferConfig` rather than having to explicitly do this?